cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/KafkaCruiseControlServletUtils.java

```diff
@@ -42,7 +42,7 @@ static String getClientIpAddress(HttpServletRequest request) {
     return request.getRemoteAddr();
   }
 
-  static String getProposalSummary(GoalOptimizer.OptimizerResult result) {
+  static String getProposalSummary(GoalOptimizer.OptimizerResult result, boolean json) {
```

Comment:
In case `json=true`, rather than returning a combined string, can we return a `Map<String, Object>` containing separate entries for:
1. `numReplicaMovements`
2. `dataToMove`
3. `numLeaderMovements`
4. `recentWindows`
5. `monitoredPartitionsPercentage`

-- i.e. we can create a distinct function for this --> `public Map<String, Object> getJsonStructure()`